osquery/sql/virtual_table.cpp

```diff
@@ -74,6 +73,194 @@ static inline std::string opString(unsigned char op) {
   return "?";
 }
 
+/// A list of tables that come from extensions; it is used to determine which
+/// table can be read/write
+std::vector<std::string> extension_table_list;
+Mutex extension_table_list_mutex;
+
+bool isExtensionTable(const std::string& table_name) {
+  ReadLock lock(extension_table_list_mutex);
+  return (std::find(extension_table_list.begin(),
+                    extension_table_list.end(),
+                    table_name) != extension_table_list.end());
+}
+
+namespace {
+// A map containing an sqlite module object for each virtual table
+std::unordered_map<std::string, struct sqlite3_module> sqlite_module_map;
+Mutex sqlite_module_map_mutex;
+
+bool getColumnValue(std::string& value,
+                    int index,
+                    int argc,
+                    sqlite3_value** argv) {
+  value.clear();
+
+  if (index >= argc) {
+    return false;
+  }
+
+  auto sqlite_value = argv[index];
+  switch (sqlite3_value_type(sqlite_value)) {
+  case SQLITE_INTEGER: {
+    auto temp = sqlite3_value_int64(sqlite_value);
+    value = std::to_string(temp);
+    break;
+  }
+
+  case SQLITE_FLOAT: {
+    auto temp = sqlite3_value_double(sqlite_value);
+    value = std::to_string(temp);
+    break;
+  }
+
+  case SQLITE_BLOB:
+  case SQLITE3_TEXT: {
+    auto data_pointer = sqlite3_value_blob(sqlite_value);
+
+    auto buffer_size = sqlite3_value_bytes(sqlite_value);
+    value.resize(buffer_size);
+    std::memcpy(&value[0], data_pointer, buffer_size);
+
+    break;
+  }
+
+  case SQLITE_NULL: {
+    break;
```

Comment:
Should we put in the `value` something like "0" or similar in this case?

Reply:
The `value` variable is cleared before we enter the switch, so in case of a NULL we get an empty string. This is how NULL values are generally handled by tables